### PR TITLE
feat: add endpoints for listing project members and deleting them

### PR DESCRIPTION
**Pull request by jan-zabloudil**
To keep PR smaller, other endpoints will be added in separate PRs.

### CODE REVIEW

File: service/project.go
Comment (bolshakoff) [inline service/project.go:412]: The user can get deleted between these two calls.

So, while I really like that it is plain and obvious, and quite smooth, here we need to add some sharpness - basically fix the race condition above.
If the user that is being deleted does not exist, there should be a corresponding pg error.

File: service/project.go
Comment (bolshakoff) [inline service/project.go:400]: Actually, a similar thing here.
Meanwhile, the project might get deleted, soo this check is not sharp enough.

In the case of member deletion, I think everything can be simplified to one call and returning a `ProjectMemberNotFoundError`.

But when it comes to listing.. see next comment...

File: service/project.go
Comment (bolshakoff) [inline service/project.go:386]: When it comes to listing, perhaps a more correct implementation here is to query the members, and if it is empty check whether it is because the project does not exist (`404`) or simply there are indeed no members in the project (`200`).

Comment (bolshakoff) [reply service/project.go:386]: But it's a minor thing; and if it is in more places, I'd maybe leave it to a separate PR or maybe issue, because the opposite argument here is also that it's such a rare and non-critical case, that not worth it...

**Review by bolshakoff — APPROVED** [review]

**Comment by jan-zabloudil** [comment]
Thanks, good point! I’ve created issue #47, where I suggested a method to prevent the race condition. Please take a look.

I will implement the fix from #47 once all open PRs are merged. If you notice this race condition occurring with other PRs as well, there's no need to mention it, as it will be addressed.

**Pull request merged**
